xarray/core/dataset.py

```diff
@@ -319,6 +322,40 @@ def load(self):
 
         return self
 
+    def compute(self):
+        """Manually trigger loading of this dataset's data from disk or a
+        remote source into memory and return a new dataset. The original is
+        left unaltered.
+
+        Normally, it should not be necessary to call this method in user code,
+        because all xarray functions should either work on deferred data or
+        load data automatically. However, this method can be necessary when
+        working with many file objects on disk.
+        """
+        # Can't just do the below, because new.variables[k] is the
+        # same object as self.variables[k]!
+        # new = self.copy(deep=False)
```

Comment:
It's done. However please see the caveat I wrote in the test script.

Further explanation to what I mean:
```
import numpy as np
from xarray import Variable, IndexVariable
from xarray.test import source_ndarray

def test_copy(data, cls):
    data = np.array(data)
    a = cls('x', data)
    print source_ndarray(data) is source_ndarray(a.data)
    b = a.copy(deep=False)
    print source_ndarray(a.data) is source_ndarray(b.data)
    
test_copy([1,2,3], Variable)
test_copy(['a', 'b', 'c'], Variable)
test_copy([1,2,3], IndexVariable)
test_copy(['a', 'b', 'c'], IndexVariable)
```
Output:
True
True
True
True
True
True
False
False